### PR TITLE
[FrameworkBundle] Allow XML configuration in config/packages by default

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | follow up of #44503
| License       | MIT
| Doc PR        | TODO

### CODE REVIEW

**Comment by derrabus** [comment]
Let me fix that table for you. 🙃

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php [hunk 1]
```diff
@@ -50,8 +50,8 @@ private function configureContainer(ContainerConfigurator $container, LoaderInte
     {
         $configDir = $this->getConfigDir();
 
-        $container->import($configDir.'/{packages}/*.{php,yaml}');
-        $container->import($configDir.'/{packages}/'.$this->environment.'/*.{php,yaml}');
+        $container->import($configDir.'/{packages}/*.{php,xml,yaml}');
+        $container->import($configDir.'/{packages}/'.$this->environment.'/*.{php,xml,yaml}');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php]: while at it: php should be last imho (the most dynamic takes over the others)

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php [hunk 2]
```diff
@@ -74,8 +74,8 @@ private function configureRoutes(RoutingConfigurator $routes): void
     {
         $configDir = $this->getConfigDir();
 
-        $routes->import($configDir.'/{routes}/'.$this->environment.'/*.{php,yaml}');
-        $routes->import($configDir.'/{routes}/*.{php,yaml}');
+        $routes->import($configDir.'/{routes}/'.$this->environment.'/*.{php,xml,yaml}');
+        $routes->import($configDir.'/{routes}/*.{php,xml,yaml}');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:78]: for routes, php should be first, so keep as is

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php]: you say i should put .xml last for BC loading order? then routes should follow isnt it.

I wasnt sure this matters from the glob pattern POV :/ but havent tested it either.

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php]: the list should be `yaml,xml,php` for services, and `php,xml,yaml` for routes yes

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php [hunk 3]
```diff
@@ -74,8 +74,8 @@ private function configureRoutes(RoutingConfigurator $routes): void
...
 
         if (is_file($configDir.'/routes.yaml')) {
```
Comment (HypeMC) [inline src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:80]: If I'm not mistaking shouldn't this case be covered as well?

Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:80]: new feature IMHO, previous PR explicitly targeted config/packages. I follow :)

changelog updated :+1:

Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:80]: ~new feature IMHO, previous PR explicitly targeted config/packages. I follow :)~

~changelog updated +1~

sorry, i missed the else case.

@nicolas-grekas WDYT?

Comment (ro0NL) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:80]: ideally i'd like this to be simple as `$routes->import($configDir.'/routes.{php,xml,yaml}');`

for services we could phase out `services_<env>.ext` in favor `when@<env>` notation IIUC.

Then everything is uniform IMHO.

**Comment by ro0NL** [comment]
i think @HypeMC's comment is valid, and this PR is a rabbit-hole :)

i may very vell rewrite xml to php once we upgrade to 6.x :} thus not bothering you with out-of-the-box xml support or so.

**Comment by fabpot** [comment]
Maybe it’s time to deprecate XML support?

**Comment by ro0NL** [comment]
@fabpot i let you decide .. maybe we miss its purpose :)

generally we could have many formats, personally i'd aim for a more consise solution, eg.

```php
$c->import($configDir.'/services.{php,xml,yaml}');
$c->import($configDir.'/{packages}/*.{php,xml,yaml}');
$c->import($configDir.'/{packages}/'.$this->environment.'/*.{php,xml,yaml}');

$r->import($configDir.'/routes.{php,xml,yaml}');
$r->import($configDir.'/routes/<env>/*.{php,xml,yaml}');
```

- no special `services_<env>.ext` file
- easy format maintenance at the glob pattern level, in alpha order :shrug: 

Then, to make it waterproof  we could forbid `some.php + some.yaml` :shrug:

**Pull request closed without merging**